### PR TITLE
FEATURE: Add Revise... option for queued post reviewable

### PR DESCRIPTION
**Pull request by martin-brennan**
This commit adds a new Revise... action that can be taken
for queued post reviewables. This will open a modal where
the user can select a Reason from a preconfigured list
(or by choosing Other..., a custom reason) and provide feedback
to the user about their post.

The post will be rejected still, but a PM will also be sent to
the user so they have an opportunity to improve their post when
they resubmit it.

image

image

image

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/modal/revise-and-reject-post-reviewable.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  @tracked reason = null;
+  @tracked customReason = null;
+  @tracked feedback = null;
+  @tracked submitting = false;
+
+  get configuredReasons() {
+    return this.siteSettings.reviewable_revision_reasons
+      .split("|")
+      .concat([OTHER_REASON]);
+  }
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/components/modal/revise-and-reject-post-reviewable.js:21]:
```suggestion
  get configuredReasons() {
    return this.siteSettings.reviewable_revision_reasons
      .split("|")
      .filter(Boolean)
      .concat([OTHER_REASON]);
  }
```

```javascript
"".split("|").concat(["no lavomatic"]) // ['', 'no lavomatic']
```

File: app/assets/javascripts/discourse/app/components/modal/revise-and-reject-post-reviewable.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        revise_feedback: this.feedback,
+      })
+      .then(() => {
+        this.submitting = false;
+        this.args.closeModal();
+      })
+      .finally(() => {
+        this.submitting = false;
+      });
+  }
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/components/modal/revise-and-reject-post-reviewable.js:51]: Could be nice to use async/await with try/catch for new code

File: app/assets/stylesheets/common/base/revise-and-reject-post-reviewable.scss [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    .post-topic .title-text {
+      font-size: var(--font-up-1);
+    }
+
+    .post-body {
+      margin: 0;
+
+      p {
+        margin: 0;
+      }
```
Comment (jjaffeux) [inline app/assets/stylesheets/common/base/revise-and-reject-post-reviewable.scss:43]: This default margin on p is hurting us more often than helping :(

File: app/assets/javascripts/discourse/app/components/modal/revise-and-reject-post-reviewable.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import { isEmpty } from "@ember/utils";
+
+const OTHER_REASON = "Other...";
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/components/modal/revise-and-reject-post-reviewable.js:7]: So this is not translated?

File: app/assets/javascripts/discourse/app/components/modal/revise-and-reject-post-reviewable.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+export default class ReviseAndRejectPostReviewable extends Component {
+  @service siteSettings;
+
+  @tracked reason = null;
+  @tracked customReason = null;
+  @tracked feedback = null;
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/components/modal/revise-and-reject-post-reviewable.js:14]: Super detail, but you don't need null here

**Review by jjaffeux — APPROVED** [review]

File: app/assets/javascripts/discourse/app/components/modal/revise-and-reject-post-reviewable.js (hunk 1 above)
Comment (martin-brennan) [reply app/assets/javascripts/discourse/app/components/modal/revise-and-reject-post-reviewable.js:21]: I appreciate the laundromat jokes in review :laughing: Thanks :+1:

File: app/assets/javascripts/discourse/app/components/modal/revise-and-reject-post-reviewable.js (hunk 2 above)
Comment (martin-brennan) [reply app/assets/javascripts/discourse/app/components/modal/revise-and-reject-post-reviewable.js:51]: Oh yep sure, will change :+1:

File: app/assets/javascripts/discourse/app/components/modal/revise-and-reject-post-reviewable.js (hunk 4 above)
Comment (martin-brennan) [reply app/assets/javascripts/discourse/app/components/modal/revise-and-reject-post-reviewable.js:7]: Actually you bring up a good point, it is a "special" reason that always exists outside the site setting but there is no reason not to have a translated version in the dropdown. Will fix

**Comment by martin-brennan** [comment]
Thank you for your review @jjaffeux :heart:

**Comment by TheIndra55** [comment]
https://github.com/discourse/discourse/blob/146da75fd703aeda466356d0736aac6d3591e0c2/config/locales/server.en.yml#L3024-L3024

Does replying to the system message actually get send to the moderation? Or is that an oversight in the message?

**Pull request merged**